composer/trainer/trainer.py

```diff
@@ -413,8 +446,11 @@ def __init__(
 
         # save_checkpoint
         save_folder: Optional[str] = None,
-        save_interval: Union[str, int, Time] = "1ep",
-        save_compression: Optional[str] = None,
+        save_name_format_string: str = "ep{epoch}-ba{batch}/rank_{rank}",
+        save_latest_symlink_format_string: str = "latest/rank_{rank}",
+        save_overwrite: bool = False,
+        should_save: Union[str, int, Time, Callable[[State, Event], bool]] = "1ep",
```

Comment:
This parameter name is kind of confusing since it's both a "should_save" and a "save_interval". Don't really have a better name top-of-mind though.

Reply:
Named to `save_interval` per https://github.com/mosaicml/composer/pull/687#discussion_r822806638